File: api-scala/src/test/scala/com/ultreon/craft/scala/testmod/init/ModItems.scala
package com.ultreon.craft.scala.testmod.init

import com.ultreon.craft.item.Item
import com.ultreon.craft.registry.{DeferRegistry, Registries, DeferredElement}
import com.ultreon.craft.scala.testmod.Constants

object ModItems {
  private final val REGISTER = DeferRegistry.of(Constants.MOD_ID, Registries.ITEM)

  final val TEST_ITEM: DeferredElement[Item] = REGISTER.defer("test_item", { () =>
    new Item(new Item.Properties)
  })

  def register(): Unit = {
    REGISTER.register()
  }
}

File: testmod/src/main/java/io/github/xypercode/init/ModItems.java
package io.github.xypercode.init;

import com.ultreon.craft.item.Item;
import com.ultreon.craft.registry.DeferRegistry;
import com.ultreon.craft.registry.Registries;
import com.ultreon.craft.registry.DeferredElement;
import io.github.xypercode.TestMod;

public class ModItems {
    private static final DeferRegistry<Item> REGISTER = DeferRegistry.of(TestMod.MOD_ID, Registries.ITEM);

    public static final DeferredElement<Item> TEST_ITEM = REGISTER.defer("test_item", () -> new Item(new Item.Properties()));

    public static void register() {
        REGISTER.register();
    }
}
